src/Corax/Queries/TermProvider.Wildcard.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using Sparrow.Server;
+using Voron;
+using Voron.Data.CompactTrees;
+
+namespace Corax.Queries
+{
+    public unsafe struct PrefixWildcardTermProvider : ITermProvider
```

Comment:
*Very* misleading name here.
There is a huge difference between prefix & contains, which is what you are actually doing.